provider/azure/cache.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"time"
+)
+
+// zonesCache is a cache for Azure zones(private or public)
+type zonesCache[T any] struct {
+	age      time.Time
+	duration time.Duration
+	zones    []T
+}
+
+// Reset method to reset the zones and update the age
+func (z *zonesCache[T]) Reset(zones []T) {
+	if z.duration > time.Duration(0) {
+		z.age = time.Now()
+		z.zones = zones
+	}
+}
+
+// Get method to retrieve the cached zones
```

Comment:
comments should add more value than writing what you can read.
In this example you can add if it's a shallow copy or not for example.

Reply:
Have added more concrete lines. Thanks